CHANGELOG.md

```diff
@@ -41,6 +41,15 @@ and this project adheres to
 
 ### Fixed
 
+- [4526](https://github.com/firecracker-microvm/firecracker/pull/4526): Added a
```

Comment:
nit: what if we rearrange sentences to tell the following story like:
- we introduced a check (summary)
- this is what happens if it's not there (if guest sends a malformed request, we copy and use host memory)
- that's how it fixes the problem (we check for max frame, drop packet and update metrics)

?

Reply:
Done!